processes/bootstrap/plugin.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2019 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bootstrap
+
+import (
+	"github.com/ligato/cn-infra/infra"
+	pm "github.com/ligato/cn-infra/processes/processmanager"
+	"github.com/ligato/cn-infra/processes/processmanager/status"
+	"github.com/pkg/errors"
+	"os"
+	"sync"
+)
+
+// Bootstrap defines methods to obtain information about running instances
+type Bootstrap interface {
+	// GetProcessNames returns names of all running process instances
+	GetProcessNames() []string
+	// GetProcessByName returns an instance of given process
+	GetProcessByName(name string) pm.ProcessInstance
+}
+
+// Plugin is a bootstrap plugin structure
+type Plugin struct {
+	mx sync.Mutex
```

Comment:
The mutexes should generally be declared next to variables they protect, in this case just before `instances` variable.